app/src/main/java/net/gsantner/markor/ui/NewFileDialog.java

```diff
@@ -171,6 +177,11 @@ private AlertDialog.Builder makeDialog(final File basedir, final boolean allowCr
                     final byte[] templateContents = getTemplateContent(templateSpinner, basedir, f.getName(), encryptCheckbox.isChecked());
                     shareUtil.writeFile(f, false, (arg_ok, arg_fos) -> {
                         try {
+                            if (appSettings.getNewFileDialogLastUsedUtf8Bom()) {
+                                arg_fos.write(0xEF);
+                                arg_fos.write(0xBB);
+                                arg_fos.write(0xBF);
+                            }
                             if (templateContents != null && (!f.exists() || f.length() < ShareUtil.MIN_OVERWRITE_LENGTH)) {
                                 arg_fos.write(templateContents);
                             }
```

Comment:
We should get wrid of own file writing implementation here and instead use FileUtils, what is used at editor as well.

But that is a topic of it's own.